src/query/execute_schema.c

```diff
@@ -13260,10 +13297,18 @@ do_recreate_att_constraints (MOP class_mop, SM_CONSTRAINT_INFO * constr_info_lis
     {
       if (SM_IS_CONSTRAINT_INDEX_FAMILY ((SM_CONSTRAINT_TYPE) constr->constraint_type))
 	{
+	  SM_INDEX_STATUS old_status = SM_NORMAL_INDEX;
+
+	  error = sm_get_constr_index_status (class_mop, constr->name, &old_status);
```

Comment:
please find the caller of the function: `do_alter_clause_change_attribute` which drops the constraint and then bring it back. it won't work.